test/projection.jl

```diff
@@ -110,17 +110,22 @@ Base.zero(x::Dual) = Dual(zero(x.value), zero(x.partial))
         @test_throws DimensionMismatch ProjectTo([1])(3.14 + im) # other array projectors don't accept numbers
         @test_throws DimensionMismatch ProjectTo(hcat([1, 2]))(3.14)
         @test pzed isa ProjectTo{AbstractArray}
+    end
 
+    @testset "Base: Ref" begin
         pref = ProjectTo(Ref(2.0))
-        @test pref(Ref(3 + im))[] === 3.0
-        @test pref(4)[] === 4.0  # also re-wraps scalars
-        @test pref(Ref{Any}(5.0)) isa Base.RefValue{Float64}
+        @test pref(Ref(3 + im)).x === 3.0
+        @test pref(4).x === 4.0  # also re-wraps scalars
+        @test pref(Ref{Any}(5.0)) isa Tangent{<:Base.RefValue}
         pref2 = ProjectTo(Ref{Any}(6 + 7im))
-        @test pref2(Ref(8))[] === 8.0 + 0.0im
+        @test pref2(Ref(8)).x === 8.0 + 0.0im
 
         prefvec = ProjectTo(Ref([1, 2, 3 + 4im]))  # recurses into contents
-        @test prefvec(Ref(1:3)) isa Base.RefValue{Vector{ComplexF64}}
+        @test prefvec(Ref(1:3)).x isa Vector{ComplexF64}
         @test_throws DimensionMismatch prefvec(Ref{Any}(1:5))
+
+        @test ProjectTo(Ref(true)) isa ProjectTo{NoTangent}
+        @test_broken ProjectTo(Ref([false]')) isa ProjectTo{NoTangent}
```

Comment:
Wasn't sure this needs fixing, the shortcut present for Diagonal has a "TODO" to remove it... but: 
```
julia> ProjectTo([false]')
ProjectTo{Adjoint}(parent = ProjectTo{NoTangent}(),)

julia> ans([1 2 3])
ERROR: type NamedTuple has no field axes

julia> ProjectTo(Diagonal([true, false]))
ProjectTo{NoTangent}()
```